### ISSUE TITLE
Eval broken for a pkgname with multiple names

### ISSUE THREAD

**Issue by stevenhoneyman** [ava1ar/customizepkg#19]
I'll try and get something in place to fix as soon as I can, but I've had very little time to spend on open source stuff lately. My current broken package is qemu. this breaks the script at line 149:

```sh
pkgname=('qemu'
         'qemu-block-iscsi'
         'qemu-block-rbd'
         'qemu-block-gluster'
         'libcacard')
```

**Comment by ava1ar** [comment]
What customizations are you applying for qemu (just want to reproduce the issue)?

**Comment by stevenhoneyman** [comment]
Any change should trigger it, but here's my slightly out-of-date config anyway:

```
remove#makedepends#libpulse
remove#makedepends#libcacard
##remove#makedepends#libseccomp
remove#makedepends#seabios
remove#makedepends#bluez-libs
remove#makedepends#spice
remove#makedepends#spice-protocol

remove#global#'libpulse'
remove#global#'seabios'
remove#global#'bluez-libs'
remove#global#'spice'
remove#global#'libcacard'
##remove#global#'libseccomp'
replace#global#!strip#strip

replace#global#--audio-drv-list='pa alsa sdl'#--audio-drv-list='alsa sdl'
replace#global#--enable-docs#--disable-docs
##replace#global#--enable-seccomp#--disable-seccomp
replace#global#--enable-spice#--disable-spice

replace#global#--enable-tpm #&--enable-{pie,curses,user,sdl,vnc} --disable-{werror,libnfs,smartcard-nss,guest-agent,libssh2,vhdx,quorum,bluez,xen,debug-info} --target-list="i386-softmmu x86_64-softmmu arm-softmmu arm-linux-user" 

replace#global#rm -r "${pkgdir}\/var"#rm -rf "${pkgdir}\/var"
removeline#global#rm....pkgdir.\/usr\/share\/qemu
```

I think the solution is going to involve changing that grep command to a small awk script, but maybe we can get away with using a PCRE instead.

**Comment by stevenhoneyman** [comment]
August... wow, that's gone quickly!

How do you like the look of this as a fix for this issue? My PCRE skills are incredibly rusty (and basic to begin with) but this seems to do the job and hasn't broken anything else yet. Maybe test for a couple of weeks before merging?

```diff
--- /bin/customizepkg	2015-12-27 21:42:26.000000000 +0000
+++ ./customizepkg	2015-12-27 21:37:33.538483104 +0000
@@ -167 +167 @@
-eval $(grep -Po '^[[:blank:]]*_?(pkg.*|name)=[^#]*(?= *#|$)' ./PKGBUILD | grep -Ev '\$\(|`')
+eval $(grep -Pzo '^[[:blank:]]*_?(pkg.*|name)=(\((.|\n)*?\)|[^#]*?$)' ./PKGBUILD | grep -Ev '\$\(|`')
```

**Comment by ava1ar** [comment]
I tried new varian on my cistomizations - looks like it works good. Regarding PCRE, my skills are also very limited, but we can always revert the change if it breaks something for somebody. Merging the fix.

**Comment by ava1ar** [comment]
Actual again: due to changes in grep behavior fix with "-Pz" options won't work anymore.

**Comment by stevenhoneyman** [comment]
How about
````diff
--- customizepkg	2016-04-03 15:53:58.904323720 +0100
+++ /bin/customizepkg	2016-04-03 15:54:34.274324246 +0100
@@ -167 +167 @@
-eval $(grep -Pzo '^[[:blank:]]*_?(pkg.*|name)=(\((.|\n)*?\)|[^#]*?(?= *#|$))' ./PKGBUILD | grep -Ev '\$\(|`')
+eval $(grep -Pzo '[^[:print:]][[:blank:]]*_?(pkg.*|name)=(\((.|\n)*?\)|[^#]*?(?= *#|\x0a))' ./PKGBUILD | grep -Ev '\$\(|`')
````

Please note that it requires an upstream bugfix to work: http://git.savannah.gnu.org/cgit/grep.git/commit/?id=d97cfd43d091662065482a8b72f17e25e296d15b

**Comment by stevenhoneyman** [comment]
Tiny addition needed (ascii files), other than this, no problems in the past month:
````diff
--- customizepkg	2016-04-28 17:59:51.025211419 +0100
+++ /bin/customizepkg	2016-04-28 18:01:42.078546397 +0100
@@ -167 +167 @@
-eval $(grep -Po '^[[:blank:]]*_?(pkg.*|name)=[^#]*(?= *#|$)' ./PKGBUILD | grep -Ev '\$\(|`')
+eval $(grep -Pazo '[^[:print:]][[:blank:]]*_?(pkg.*|name)=(\((.|\n)*?\)|[^#]*?(?= *#|\x0a))' ./PKGBUILD | grep -Eva '\$\(|`')
````

**Comment by ava1ar** [comment]
@stevenhoneyman Did the bugix for grep you mentioned already released?

**Comment by stevenhoneyman** [comment]
Yes, it's included in v2.25, released 9 days ago.

**Comment by ava1ar** [comment]
Merged the fix. Thanks, @stevenhoneyman!
